tripy/examples/resent50/resnet50.ipynb

```diff
@@ -0,0 +1,529 @@
+   "source": [
+    "class ResNetClassifier(tp.Module):\n",
+    "    def __init__(self, num_classes=1000):\n",
+    "        super(ResNetClassifier, self).__init__()\n",
+    "        self.resnet = ResNetModel()\n",
+    "        self.classifier = tp.Linear(2048, num_classes)\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        features = self.resnet(x)\n",
+    "        features = tp.flatten(features, start_dim=1)\n",
+    "        output = self.classifier(features)\n",
+    "        return output"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Instantiate the Tripy model\n",
+    "tripy_model = ResNetClassifier()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load weights to Tripy Model\n",
+    "\n",
+    "First we ensure all state dict keys match and then we load the pretrained weights."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_pretrained_weights_tp(model, pretrained_state_dict):\n",
+    "    \"\"\"\n",
+    "    This function loads weights from a state_dict into the custom TPResNet model,\n",
+    "    converting them into tp.Parameter before loading.\n",
+    "    \"\"\"\n",
+    "    model_dict = model.state_dict()  # Get the model's state dict\n",
+    "\n",
+    "    # Convert each weight to a tp.Parameter and update the model's state dict\n",
+    "    for k, v in pretrained_state_dict.items():\n",
+    "        if 'num_batches_tracked' in k:\n",
+    "            continue    # Skip num_batches_tracked since it is not needed for inference (tp.BatchNorm does not support)\n",
+    "        if 'classifier' in k:\n",
+    "            k = k.replace('1.', '') # Remove extra 1 (slight naming mismtach)\n",
+    "\n",
+    "        assert k in set(model_dict.keys())\n",
+    "        new_v = tp.Parameter(v.contiguous())\n",
+    "        model_dict[k] = new_v \n",
+    "\n",
+    "    model.load_state_dict(model_dict)\n",
+    "    return model\n",
+    "\n",
+    "pretrained_state_dict = resnet_pretrained.state_dict()\n",
+    "tripy_model = load_pretrained_weights_tp(tripy_model, pretrained_state_dict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we compile the model with static input shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_shape = [1, 3, 224, 224]\n",
+    "compiled_model = tp.compile(tripy_model, args=[tp.InputInfo(input_shape, dtype=tp.float32)])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run a dummy forward path\n",
+    "x = tp.ones(input_shape, dtype=tp.float32)\n",
+    "compiled_output = compiled_model(x)\n",
+    "compiled_output.shape"
```

Comment:
What's the reason for this dummy forward pass?

Reply:
just to demonstrate that compiled model works. Can remove it, since we actually do the demo down below.